src/core/Form/TextInput/TextInput.baseStyles.tsx

```diff
@@ -1,40 +1,59 @@
 import { css } from 'styled-components';
+import { TextInputProps } from './TextInput';
 import { withSuomifiTheme, TokensAndTheme } from '../../theme';
 import { input, inputContainer, font } from '../../theme/reset';
 import { math } from 'polished';
 
 export const baseStyles = withSuomifiTheme(
-  ({ theme }: TokensAndTheme) => css`
+  ({
+    theme,
+    inputContainerProps,
+    fullWidth,
+  }: TokensAndTheme & Omit<TextInputProps, 'labelText'>) => css`
+  
+  &.fi-text-input {
+    display: inline-block;
+    width: ${
+      fullWidth
+        ? '100%'
+        : inputContainerProps?.style?.width
+        ? inputContainerProps?.style.width
+        : inputContainerProps?.width || '290px'
+    };
+  }
+
   & .fi-text-input_label-p {
     margin-bottom: ${theme.spacing.xs};
     ${font({ theme })('actionElementInnerTextBold')};
     color: ${theme.colors.blackBase};
   }
 
-  & .fi-text-input_container {
+  & .fi-text-input_input-element-container {
     ${inputContainer({ theme })}
   }
 
   & .fi-text-input_statusText_container {
     display: flex;
     flex-direction: column;
+    font-size: 14px;
```

Comment:
Moving this here seems to result in hint text having wrong text size. It gets 16px as the size from somewhere.

Reply:
Good catch! Fixed!